src/bootstrap/native.rs

```diff
@@ -170,12 +169,13 @@ impl Step for Llvm {
         //
         // If we are shipping llvm tools then we statically link them LLVM
         if (target.contains("linux-gnu") || target.contains("apple-darwin")) &&
-            !builder.config.llvm_tools_enabled {
+            !builder.config.llvm_tools_enabled &&
+            !builder.config.lldb_enabled {
```

Comment:
Hm this could be a bit worrisome, this may affect how we can distribute LLVM? If we have LLVM linked dynamically we're not shipping the various libraries for the tools (LLD and such) as well as rustc.

Reply:
The comment just above this says that the intent is to ensure that everything is statically linked against LLVM when the tools are being distributed.  So, this addition just ensures that the same is true for lldb.  That is, if the tools (llvm's objdump, etc; and now including lldb) are being built, then disable dynamic linking.  Otherwise, enable it.

Reply:
Ah right yes sorry, I think I misread the conditionals here!